src/target.d

```diff
@@ -153,7 +153,12 @@ struct Target
      */
     extern (C++) static uint fieldalign(Type type)
     {
-        return type.alignsize();
+        const size = type.alignsize();
+
+        if ((global.params.is64bit || global.params.isOSX) && (size == 16 || size == 32))
+            return size;
+
+        return (8 < size) ? 8 : size;
```

Comment:
Meh, I don't use this code. It doesn't make sense to me either.  You shouldn't be using this code either.